examples/web-chat/src/App.tsx

```diff
@@ -1,230 +1,51 @@
-import { useEffect, useReducer, useState } from "react";
 import "./App.css";
 import handleCommand from "./command";
 import Room from "./Room";
-import { WakuContext } from "./WakuContext";
-import { ThemeProvider } from "@livechat/ui-kit";
-import { generate } from "server-name-generator";
 import { Message } from "./Message";
-import { wakuDnsDiscovery } from "@waku/dns-discovery";
-import { wakuPeerExchangeDiscovery } from "@waku/peer-exchange";
-import { waitForRemotePeer } from "@waku/core";
-import { Protocols, WakuLight } from "@waku/interfaces";
-import { createLightNode } from "@waku/create";
-import { DecodedMessage, Decoder } from "@waku/core/lib/message/version_0";
-import { PageDirection } from "@waku/interfaces";
+import { PageDirection, LightNode } from "@waku/interfaces";
 
-const themes = {
-  AuthorName: {
-    css: {
-      fontSize: "1.1em",
-    },
-  },
-  Message: {
-    css: {
-      margin: "0em",
-      padding: "0em",
-      fontSize: "0.83em",
-    },
-  },
-  MessageText: {
-    css: {
-      margin: "0em",
-      padding: "0.1em",
-      paddingLeft: "1em",
-      fontSize: "1.1em",
-    },
-  },
-  MessageGroup: {
-    css: {
-      margin: "0em",
-      padding: "0.2em",
-    },
-  },
-};
-
-export const ChatContentTopic = "/toy-chat/2/huilong/proto";
-const ChatDecoder = new Decoder(ChatContentTopic);
-
-async function retrieveStoreMessages(
-  waku: WakuLight,
-  setArchivedMessages: (value: Message[]) => void
-): Promise<void> {
-  const startTime = new Date();
-  // Only retrieve a week of history
-  startTime.setTime(Date.now() - 1000 * 60 * 60 * 24 * 7);
-
-  const endTime = new Date();
+import { useWaku, useContentPair } from "@waku/react";
 
-  try {
-    for await (const messagesPromises of waku.store.queryGenerator(
-      [ChatDecoder],
-      {
-        pageSize: 5,
-        pageDirection: PageDirection.FORWARD,
-        timeFilter: {
-          startTime,
-          endTime,
-        },
-      }
-    )) {
-      const wakuMessages = await Promise.all(messagesPromises);
+import { useMessages, usePersistentNick } from "./hooks";
 
-      const messages: Message[] = [];
-      wakuMessages
-        .filter(isMessageDefined)
-        .map((wakuMsg) => Message.fromWakuMessage(wakuMsg))
-        .forEach((message) => {
-          if (message) {
-            messages.push(message);
-          }
-        });
-      setArchivedMessages(messages);
-    }
-  } catch (e) {
-    console.log("Failed to retrieve messages", e);
-  }
-}
+const startTime = new Date();
```

Comment:
Future improvement: instead of using start/end time. Do a backward query and stop it once either:
- message retrieve is more than a week old
- Or retrieved 100 msg

Reply:
will include into follow up